definitions/procedures/backup/prepare_directory.rb

```diff
@@ -21,8 +21,10 @@ def run
       end
 
       FileUtils.rm(Dir.glob(File.join(@backup_dir, '.*.snar'))) if @preserve_dir
-      if @incremental_dir
-        FileUtils.cp(Dir.glob(File.join(@incremental_dir, '.*.snar')), @backup_dir)
+      if @incremental_dir && !(snar_files = Dir.glob(File.join(@incremental_dir, '.*.snar'))).empty?
+        FileUtils.cp(snar_files, @backup_dir)
+      else
+        raise 'No .snar files found in previous backup directory'
```

Comment:
As there are two possibilities, I think we should change error message. Something like -
```suggestion
        raise "May be either incremental backup directory or details such as .snar files under this directory is missing. Could you please provide a valid previous backup directory."
```
@upadhyeammit, any suggestions on message wording?

Reply:
Maybe:

"Either #{@incremental_dir} or #{@incremental_dir}/*.snar files unavailable. Provide valid previous backup directory"